### PR TITLE
Use graphql instead of json

### PR DESCRIPTION
**Pull request by jyc979**
In regard to #94 , this makes it use graphql instead of json.

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/eatery/network/NetworkUtilities.java [hunk 1]
```diff
@@ -63,14 +72,30 @@ public static String getJSON() {
...
+        activity.runOnUiThread(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              ((MainActivity)activity).getSupportFragmentManager()
+                  .beginTransaction()
+                  .replace(R.id.frame_fragment_holder, new MainListFragment())
+                  .commit();
+            } catch (Exception e) {
+              Log.e(TAG, e.getMessage());
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/network/NetworkUtilities.java]: I'm not a huge fan of Log. Instead, this should be `e.printStackTrace()`

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java [hunk 2]
```diff
@@ -65,16 +65,18 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
...
 
-    new ProcessJson().execute("");
+    // Try pulling data from GraphQL, if not fallback to json from cornell dining
+    NetworkUtilities.getEateries(presenter,this);
+    if (JSON_FALLBACK) {
+      new ProcessJson().execute("");
+    }
   }
 
   public class ProcessJson extends AsyncTask<String, Void, ArrayList<EateryBaseModel>> {
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: Right now, half of the networking logic is on the activity and the other half is in a networking class. I feel like it's almost better to integrate `ProcessJson` into `NetworkUtils`.

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java [hunk 3]
```diff
@@ -65,16 +65,18 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
           }
         });
 
-    new ProcessJson().execute("");
+    // Try pulling data from GraphQL, if not fallback to json from cornell dining
+    NetworkUtilities.getEateries(presenter,this);
+    if (JSON_FALLBACK) {
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: Did you test to make sure that `ProcessJson` will be called if graphql fails?

File: app/src/main/java/com/cornellappdev/android/eatery/network/NetworkUtilities.java (hunk 1 above)
Comment (jyc979) [reply app/src/main/java/com/cornellappdev/android/eatery/network/NetworkUtilities.java]: Done

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java (hunk 2 above)
Comment (jyc979) [reply app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: Discussed privately. Resolved

File: app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java (hunk 3 above)
Comment (jyc979) [reply app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java]: Yep. Resolved.

**Pull request merged**